src/traits.rs

```diff
@@ -0,0 +1,29 @@
+use {
+    crate::{TextRange, TextSize},
+    std::convert::TryInto,
+};
+
+/// Text-like structures that have a text size.
+pub trait TextSized {
+    /// The size of this text-alike.
+    fn text_size(&self) -> TextSize;
+}
+
+impl TextSized for str {
+    fn text_size(&self) -> TextSize {
+        let len = self.len();
+        TextSize::new(len)
+    }
+}
+
+impl TextSized for char {
+    fn text_size(&self) -> TextSize {
+        self.len_utf8().try_into().unwrap()
```

Comment:
There still will be extra code in debug though? And optimization takes times. I guess we can do `TextSize(u32::from(self.len_utf()).into())`?

Reply:
The extra debug code is fair. I've already changed it to use `as`, actually, as it's trivially correct. `Into` can't be used because `usize` is not `Into<u32>` (because rustc pretends to be compatible with `pointer_width = "16" even though there are issues with that compatibility).